evo_researcher/functions/cache.py

```diff
@@ -0,0 +1,19 @@
+import os
+from typing import TypeVar, Callable, Any, cast
+from joblib import Memory
+from functools import cache
+from dotenv import load_dotenv
+
+load_dotenv()
+
+MEMORY = Memory("./.cachedir", verbose=0)
+ENABLE_CACHE = os.getenv("ENABLE_CACHE", "0") == "1"
+
+T = TypeVar("T", bound=Callable[..., Any])
+
+
+def persisted_inmemory_cache(func: T) -> T:
```

Comment:
Nit, but I think it should be `persisted_inmemory_cache` -> `persistent_inmemory_cache`

Reply:
Sure, it sounds better 👌